### PR TITLE
Closes #956 - Catches JsonSyntaxException while performing ADAL -> MSAL migration

### LINKED ISSUE #956: JsonSyntaxException while performing ADAL -> MSAL migration
[Workitem (corpnet reqd)](https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20Android/Backlog%20items/?workitem=1065467)

Minimal impact in terms of user count, but leads to a nasty blocked state.

```java
com.google.gson.JsonSyntaxException:
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read (ReflectiveTypeAdapterFactory.java:224)
at com.google.gson.Gson.fromJson (Gson.java:888)
at com.google.gson.Gson.fromJson (Gson.java:853)
at com.google.gson.Gson.fromJson (Gson.java:802)
at com.google.gson.Gson.fromJson (Gson.java:774)
at com.microsoft.identity.common.internal.migration.AdalMigrationAdapter.deserialize (AdalMigrationAdapter.java:211)
at com.microsoft.identity.common.internal.migration.AdalMigrationAdapter.adapt (AdalMigrationAdapter.java:121)
at com.microsoft.identity.common.internal.migration.TokenMigrationUtility$1.run (TokenMigrationUtility.java:67)
at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
at java.lang.Thread.run (Thread.java:919)
Caused by: java.lang.IllegalStateException:
at com.google.gson.stream.JsonReader.beginObject (JsonReader.java:385)
at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read (ReflectiveTypeAdapterFactory.java:213)
```

### PR DESCRIPTION
**Pull request by iambmelt**
Closes #956

### CODE REVIEW

File: common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                new Object[]{
+                        getAdalCacheItemInput(),
+                        getDeserializationValidator()
+                },
+                new Object[]{
+                        getMalformedCacheItemInput(),
+                        getDeserializationValidator()
+                }
+        );
+    }
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:74]: You know I love parameterized tests, but the question I would ask is that with only 2 permutations here is it clearer to just have the two methods?

Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:74]: Originally had exactly that, and expanded upon it:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/957/commits/7ad92ac9b6279811ca9e37851070a63b4fac06c4

So reverting this is trivial, your call on which you want

Comment (AdamBJohnsonx) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:74]: I'm not concerned with it, really.  An name parameter would be nice, but I doubt that your class arguments give useful toString values.

File: common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class AdalMigrationAdapterDeserializationTest {
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:46]: Does it need to be an instrumented test? Wouldn't a JVM test be more efficient?

Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:46]: Need a context

File: common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java (hunk 1 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java]: Added a name param via the 0th param: 81f02b1f98dd4f78c46c65940bd42826ad2f607f

File: common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java (hunk 2 above)
Comment (shahzaibj) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:46]: You can get a context in JVM based tests. `ApplicationProvider.getApplicationContext()` - it's an AndroidX API

All of our E2E Robolectric tests need a `context` as well and we get it like that and run the tests in JVM

Comment (shahzaibj) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java:46]: Instrumented test is fine too....it's just that we've been trying to move to Robolectric test as a team, with possibly moving our existing Instrumented test to Robolectric in the future....adding more instrumented tests would just make that more difficult to happen

Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java]: e143f6dbbdd0787210be518e3bfdda4180635ae6

**Pull request merged**